drivers/ds3231/ds3231.c

```diff
@@ -171,9 +176,25 @@ int ds3231_init(ds3231_t *dev, const ds3231_params_t *params)
         set = CTRL_EOSC;
     }
 
+    /* enables  the alarm interrupt control (no SQW output) */
+    set |= CTRL_INTCN;
+
     return _clrset(dev, REG_CTRL, clr, set, 0, 1);
 }
 
+#if IS_USED(MODULE_DS3231_INT)
+int ds3231_init_int(ds3231_t *dev, ds3231_alarm_cb_t cb, void *arg)
+{
+    assert(dev != NULL);
+    assert(gpio_is_valid(dev->int_pin));
+
+    /* SQW is set to low when an alarm is trigerred */
+    gpio_init_int(dev->int_pin, GPIO_IN, GPIO_FALLING, cb, arg);
```

Comment:
```suggestion
    /* SQW is set to low when an alarm is triggered */
    if (gpio_init_int(dev->int_pin, GPIO_IN, GPIO_FALLING, cb, arg) < 0) {
        return -EIO;
    }
```

In the rare case that `gpio_init_int` fails 😉